sdk/program/src/vote/state/mod.rs

```diff
@@ -414,13 +427,22 @@ impl VoteState {
             }
         }
 
-        self.epoch_credits.last_mut().unwrap().1 += credits;
+        self.epoch_credits.last_mut().unwrap().1 = self
```

Comment:
i'm surprised the borrow checker is cool with this. looks like two mutable references to `self.epoch_credits`.  maybe ok 'cause same statement :thinking:

Reply:
wow yeah surprised it let me get away with that, uneducated guess: maybe something to do with `saturating_add` consuming `self`?

Reply:
yeah the borrow of the right side can be dropped by the time the assignment happens